src/cards/contaminate.ts

```diff
@@ -27,65 +30,103 @@ const spell: Spell = {
     effect: async (state, card, quantity, underworld, prediction) => {
       // .filter: only target living units
       for (let unit of state.targetedUnits.filter(u => u.alive)) {
-        await spreadCurses(state.casterPlayer, unit, underworld, state.aggregator.radius, prediction);
+        contaminate(state.casterPlayer, unit, underworld, state.aggregator.radius, prediction, quantity);
```

Comment:
I believe this still must be `await`ed.  Or else if this is cast in sequence it will finish asynchronously after the next spell in the spell order instead of before it.

Reply:
Oh true! I wanted all contaminates to go at the same time, because otherwise it takes suuuper long. I need to turn this into a promise[] I think, and await full completion. Thanks for pointing that out!